src/Validator/Rules/PossibleTypeExtensions.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Validator\Rules;
+
+use GraphQL\Error\Error;
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Language\AST\Node;
+use GraphQL\Language\AST\NodeKind;
+use GraphQL\Language\AST\TypeDefinitionNode;
+use GraphQL\Language\VisitorOperation;
+use GraphQL\Type\Definition\EnumType;
+use GraphQL\Type\Definition\InputObjectType;
+use GraphQL\Type\Definition\InterfaceType;
+use GraphQL\Type\Definition\NamedType;
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\ScalarType;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Definition\UnionType;
+use GraphQL\Utils\Utils;
+use GraphQL\Validator\SDLValidationContext;
+
+/**
+ * Possible type extension.
+ *
+ * A type extension is only valid if the type is defined and has the same kind.
+ */
+class PossibleTypeExtensions extends ValidationRule
+{
+    public function getSDLVisitor(SDLValidationContext $context): array
+    {
+        $schema = $context->getSchema();
+        /** @var array<string, TypeDefinitionNode&Node> $definedTypes */
+        $definedTypes = [];
+
```

Comment:
```suggestion

        /** @var array<string, TypeDefinitionNode&Node> $definedTypes */
        $definedTypes = [];
```

Reply:
What is the suggestion here? The code is the same.

Reply:
Whitespace. I prefer to have variables that are filled in a loop as close as possible to that loop.